### PR TITLE
Add manifesto page

### PR DESCRIPTION
**Pull request by pepibumur**
### Short description 📝
As the project matures, I think it's important to have a place in the documentation where we describe the pillars to the design of Tuist. Some of those principles have been floating around, and we remind contributors about them when they open PRs, so I think we should have them written down.

This PR only contains 2 but the idea is to add 10 in total. I have al list so I just need to expand each of them. Please if there are other principles that you believe are also important, don't hesitate to share them.

### CODE REVIEW

File: website/markdown/docs/contribution/manifesto.mdx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+---
+name: Manifesto
+order: 10
+excerpt: This page describes principles that are pillars to the design and development of Tuist. They evolve over time with the project and are meant to ensure a sustainable growth that is well-aligned with the project foundation.
```
Comment (lakpa) [inline website/markdown/docs/contribution/manifesto.mdx]: principles `those` are pillars`?

File: website/markdown/docs/contribution/manifesto.mdx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+For example, there’s a convention for defining dependencies between targets by using the provided public interface. By doing that, Tuist ensures that the projects are generated with the right configurations for the linking to work. Developers have the option to define the dependencies through build settings, but they’d be doing it implicitly and therefore breaking Tuist features such as graph that rely on some conventions being followed.
+
+The reason why we default to conventions is because the more decision we can make on behalf of the developers, the more focus they’ll have crafting features for their apps. When we are left with no conventions, like it’s the case in many projects, we have to make decisions that will end up not being consistent with other decisions and as a consequence there’ll be accidental complexity that will be hard to manage.
```
Comment (lakpa) [inline website/markdown/docs/contribution/manifesto.mdx]: the more focus they’ll `be` crafting features

Comment (lakpa) [inline website/markdown/docs/contribution/manifesto.mdx]: there’ll be `an` accidental complexity

File: website/markdown/docs/contribution/manifesto.mdx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+## 2. Manifests are the source of truth
+
+Having many layers of configurations and contracts between them results in a project setup that is hard to reason about and maintain. Think for a second on an average project. The definition of the project lives in the _.xcodeproj_ directories, the CLI in scripts _(e.g Fastfiles)_, and the CI logic in pipelines. Those are three layers with contracts between them that we need to maintain. How often have you been in a situation where you changed something in your projects, and then a week later you realized that the release scripts broke?
+
+We can simplify this by having a single source of truth, the manifest files. Those files provide Tuist with the information that it needs to generate Xcode projects that developers can use to edit their files. Moreover, it allows having standard commands for building the projects from a local or CI environment.
```
Comment (lakpa) [inline website/markdown/docs/contribution/manifesto.mdx]: wondering how many manifest are we supporting per project, one?

Comment (pepibumur) [reply website/markdown/docs/contribution/manifesto.mdx]: Could you expand on this? Not sure I get the question.

**Pull request merged**